### PR TITLE
Add Newtype exercise to calculate wattage.

### PR DESCRIPTION
**Pull request by shaunplee**
One simple proposal to address #255 using an extension of the existing example in the Newtype section.

### CODE REVIEW

**Comment by milesfrain** [comment]
I think this looks good overall.
Might want to make this a bit more straightforward for readers with:
```purs
calculateWattage :: Amp -> Volt -> Watt
```

**Comment by shaunplee** [comment]
> Might want to make this a bit more straightforward for readers with:
> 
> ```purescript
> calculateWattage :: Amp -> Volt -> Watt
> ```

Yes, I like that better.

I also missed that there was already a definition of `Watt`, so I've rewritten the exercise accordingly and also rewrote the test to use the existing `MakeWatt` constructor.

File: text/chapter5.md [hunk 1]
```diff
@@ -456,6 +456,14 @@ In this case, `Watt` is the _type constructor_ and `MakeWatt` is the _data const
 
 Another application of newtypes is to attach different _behavior_ to an existing type without changing its representation at runtime. We cover that use case in the next chapter when we discuss _type classes_.
 
+## Exercises
+
+1. (Easy) Define a `calculateWattage` function using the above definitions `Amp`, `Volt`, and `Watt`:
+```haskell
+calculateWattage :: Amp -> Volt -> Watt
+```
+that calculates a wattage in Watts as the product of the given current in Amps and voltage in Volts.
```
Comment (milesfrain) [inline text/chapter5.md:465]: Don't know if this makes things any clearer. Up to your discretion. Will merge either way.
```suggestion
that calculates a wattage in `Watt`s as the product of the given current in `Amp`s and voltage in `Volt`s.
```

File: exercises/chapter5/test/Main.purs [hunk 2]
```diff
@@ -2,13 +2,13 @@ module Test.Main where
...
-import Test.NoPeeking.Solutions
 
-import ChapterExamples (Amp(..), current, fromString, gcd, gcdV2, isEmpty, livesInLA, lzs, partialFunction, showPerson, showPersonV2, sortPair, takeFive, toString, unknownPerson)
+import Test.NoPeeking.Solutions as Solutions
+import ChapterExamples (Amp(..), current, fromString, gcd, gcdV2, isEmpty, livesInLA, lzs, partialFunction, showPerson, showPersonV2, sortPair, takeFive, toString, unknownPerson, Volt(..), Watt(..))
 import Data.Int (round)
 import Data.Maybe (Maybe(Just, Nothing))
 import Data.Person (Person)
-import Data.Picture (Shape(..), Picture, getCenter, origin)
+import Data.Picture (Shape(..), Picture, getCenter, origin, shapeBounds)
```
Comment (milesfrain) [inline exercises/chapter5/test/Main.purs:11]: Let's rebase this on the fix for #274 (in progress). That should simplify some of these import diffs. There are also some complications in this PR due to this chapter is missing the automated solutions reset flag. I'll fix that as well before rebasing this PR.
```purs
import Test.NoPeeking.Solutions  -- Note to reader: Delete this line
```

**Comment by milesfrain** [comment]
Let me know if you're actively rebasing this. I was going to do that now to clean-up the merge conflict I introduced.

File: exercises/chapter5/test/Main.purs [hunk 3]
```diff
@@ -111,9 +111,8 @@ Note to reader: Delete this line to expand comment block -}
           $ shapeText $ Line origin { x: 1.0, y: 1.0 }
     suite "Exercise Group - Newtype" do
       test "Exercise - calculateWattage" do
-        let
-          (MakeWatt w) = calculateWattage (Amp 0.5) (Volt 120.0)
-        Assert.equal 60.0 w
+        Assert.equal (MakeWatt 60.0)
+          $ calculateWattage (Amp 0.5) (Volt 120.0)
```
Comment (milesfrain) [inline exercises/chapter5/test/Main.purs]: I think this is a little more idomatic

**Comment by milesfrain** [comment]
Let me know if you're ready for this PR to merge.

File: exercises/chapter5/test/Main.purs (hunk 3 above)
Comment (shaunplee) [reply exercises/chapter5/test/Main.purs]: I think I tried something like that, but as I recall, the compiler complained about the lack of an Eq instance for `Watt`. If this works, I'm good with it.

**Comment by milesfrain** [comment]
> I also missed that there was already a definition of `Watt`, so I've rewritten the exercise accordingly and also rewrote the test to use the existing `MakeWatt` constructor.

I'm thinking this exercise might be even more valuable to readers if we don't rely on an existing constructor, so they get to create their own. That would involve the following changes:
- Change the `MakeWatt` example in the text to `MakeColumb`
- Restore your manual unwrapping in the test (since we can't ask readers to derive their own `Eq` and `Show` instances yet)

This could be a follow-up PR.

Also, sorry if the force push on your branch causes any inconvenience. I find that makes rebases easier to manage (assuming you're not also actively working on the same branch).

**Comment by shaunplee** [comment]
Sorry, my builds keep failing because it seems I'm missing some step. It looks like some issue with testing the `Shape` implementation in `Picture.purs` versus `ShapeExt` implementation in `Solutions.purs` but I'm not exactly clear on how this is handled.

**Comment by milesfrain** [comment]
This is unfortunately a bit of a tricky (and rare) situation with the book's workaround for both allowing CI on "official" solutions, while also allowing resetting solutions when the exercise involves modifying an existing ADT.
Anyway, it's fixable by just avoiding the duplicate `shapeBounds` import from `Data.Picture` until the other implicit import from `Test.NoPeeking.Solutions` is cleared away during solutions reset.

This is fixed now. Awaiting your approval to merge.

**Comment by shaunplee** [comment]
@milesfrain Thanks for your help!

**Pull request merged**